lib/src/lib/components/form-fields/amount-field/amount.helpers.ts

```diff
@@ -0,0 +1,65 @@
+import { AbstractControl, ValidationErrors, ValidatorFn } from '@angular/forms';
+import { AmountOptions } from './amount-field.model';
+
+export function getSeparatorCount(value: string, separator: string): number {
+  return value?.match(new RegExp(separator, 'g'))?.length ?? 0;
+}
+
+export function amountToNumber(
+  value: string,
+  errorCb?: (error: string) => null
+): number | null {
+  if (value?.length) {
+    const commaCount = getSeparatorCount(value, ',');
+    const pointCount = getSeparatorCount(value, '\\.');
+    // the input value can only contain one separator sign
+    if (commaCount + pointCount > 1) {
+      return errorCb ? errorCb('toManyDecimalSeparators') : null;
+    }
```

Comment:
won't this throw the error when I `amountToNumber("204.500,45")` ?

Reply:
uh nice catch, sometimes I do wanna input the separators myself!

Reply:
Yes, that is the point. This validation will only happen in focused mode.

Reply:
I'm not sure how the events work but in this case if you were writing it yourself then 204 and adding 5 would already format to 2.045 right? It would be a problem when you paste, in which case you need to look for one separator of one type and 0 to n separators of another type, but max two types of separators.

Reply:
Cause you should error out in the case of `200.500 1,4` as this would be very annoying to detect

Reply:
200.500 1,4 -> will give an error if you paste this in